src/Orleans/Streams/PersistentStreams/IStreamQueueBalancer.cs

```diff
@@ -8,27 +8,32 @@ namespace Orleans.Streams
     /// The stream queue balancer is responsible for load balancing queues across all other related queue balancers.  It
     /// notifies any listeners (<code>IStreamQueueBalanceListener</code>) of changes to the distribution of queues.
     /// </summary>
-    internal interface IStreamQueueBalancer
+    public interface IStreamQueueBalancer
     {
         /// <summary>
+        /// Initialize this instance
+        /// </summary>
+        /// <returns></returns>
+        Task Initialize();
+        /// <summary>
         /// Retrieves the latest queue distribution for this balancer.
         /// </summary>
         /// <returns>Queue allocated to this balancer.</returns>
-        IEnumerable<QueueId> GetMyQueues();
+        Task<IEnumerable<QueueId>> GetMyQueues();
```

Comment:
May need to note in the comments, that GetMyQueues, SubscribeToQueueDistributionChangeEvents, and UnSubscribeFromQueueDistributionChangeEvents will likely be called in the IStreamQueueBalanceListener's thread so they need to be thread safe.